src/benchmark_ocl.cpp

```diff
@@ -144,125 +119,106 @@ int main(void)
 	const std::string compile_options_gpu = compile_options_common + " -DVEC_LENGTH=2 -DCHUNK_SIZE=" STR(CHUNK_SIZE) " -DNUM_SUB_GROUPS=" STR(NUM_SUB_GROUPS);
 
 	cl_int err = 0;
-	clu_initialize_params init_params = {}; // NOTE: null initialise!
-	if (DEVICE_TYPE != CL_DEVICE_TYPE_GPU)
-		init_params.vendor_name = "Intel(R) Corporation";
-	init_params.default_queue_props = CL_QUEUE_PROFILING_ENABLE;
-	init_params.preferred_device_type = DEVICE_TYPE; // set device type
-	init_params.compile_options = compile_options_common.c_str(); // set default compile options
-	cl_int status = cluInitialize(&init_params);
-
+	// set up OpenCL using noma-ocl
+	noma::ocl::config ocl_config("", false);
+	noma::ocl::helper ocl_helper(ocl_config);
 	// output the used device
-	cl_device_id dev_id;
-	err = clGetCommandQueueInfo(CLU_DEFAULT_Q, CL_QUEUE_DEVICE, sizeof(cl_device_id), &dev_id, nullptr);
-	ocl_error_handler(err, "clGetCommandQueueInfo()");
-	clu_device_info dev_info = cluGetDeviceInfo(dev_id, &err);
-	ocl_error_handler(err, "cluGetDeviceInfo()");
-	std::cerr << "OpenCL device name:    " << dev_info.device_name << std::endl;
-	std::cerr << "OpenCL device vendor:  " << dev_info.device_vendor << std::endl;
-	std::cerr << "OpenCL device profile: " << dev_info.device_profile << std::endl;
-	std::cerr << "OpenCL device version: " << dev_info.device_version << std::endl;
-	std::cerr << "OpenCL driver version: " << dev_info.driver_version << std::endl;
-
-	// allocate OpenCL device memory
-	cl_mem hamiltonian_ocl = clCreateBuffer(CLU_CONTEXT, CL_MEM_READ_ONLY, size_hamiltonian_byte, 0, &err);
-	ocl_error_handler(err, "clCreateBuffer(hamiltonian_ocl)");
-	cl_mem sigma_in_ocl = clCreateBuffer(CLU_CONTEXT, CL_MEM_READ_WRITE, size_sigma_byte, 0, &err);
-	ocl_error_handler(err, "clCreateBuffer(sigma_in_ocl)");
-	cl_mem sigma_out_ocl = clCreateBuffer(CLU_CONTEXT, CL_MEM_READ_WRITE, size_sigma_byte, 0, &err);
-	ocl_error_handler(err, "clCreateBuffer(sigma_out_ocl)");
+	ocl_helper.write_device_info(std::cerr);
+
+	// allocate OpenCL device memory // DONE: replace with create_buffer() from noma::ocl, use C++ interface
+	noma::ocl::buffer hamiltonian_ocl = ocl_helper.create_buffer(CL_MEM_READ_ONLY,  size_hamiltonian_byte);
+	noma::ocl::buffer sigma_in_ocl    = ocl_helper.create_buffer(CL_MEM_READ_WRITE, size_sigma_byte      );
```

Comment:
Not a big fan of this aligned style, as there are situations with very long lines with additional line breaks etc. where it does not work well, and then it cannot be applied as a general rule etc... I'm OK with it when it helps with readability a lot, but please remove at least the trailing spaces after the last argument

Reply:
Changed alignment style where it is not necessary with a758cba0fa201d64c477ff298ffa7ea784117a09.